### PR TITLE
Allow admins to limit the size of attachements

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes https://github.com/nextcloud/mail/issues/3057

Todo
- [x] Front-end size check
- [x] Back-end setting
- [x] Sane default value?

### CODE REVIEW

File: templates/index.php [hunk 1]
```diff
@@ -29,6 +29,7 @@
 ?>
 
 <input type="hidden" id="debug-mode" value="<?php p($_['debug'] ? 'true' : 'false'); ?>">
+<input type="hidden" id="attachment-size-limit" value="<?php p($_['attachment-size-limit']); ?>">
```
Comment (ChristophWurst) [inline templates/index.php:32]: we should use initial state. will send a follow-up PR

**Comment by ChristophWurst** [comment]
cc @johnnyq

**Comment by GretaD** [comment]
this error message popup everytime i want to attach smth, even if its 1kb.
error_attachment
also it says 0B, shouldnt it be 3(thats the limit right)?

File: src/components/ComposerAttachments.vue [hunk 2]
```diff
@@ -93,22 +99,58 @@ export default {
...
 		onLocalAttachmentSelected(e) {
 			this.uploading = true
 
 			Vue.set(this, 'uploads', {})
 
+			const toUpload = Object.values(e.target.files).reduce((acc, file) => {
+				return acc + file.size
+			}, 0)
+			const newTotal = toUpload + this.totalSizeOfUpload()
+			logger.debug('checking upload size limit', {
```
Comment (ChristophWurst) [inline src/components/ComposerAttachments.vue:133]: @GretaD what does this line log for you?

Comment (GretaD) [reply src/components/ComposerAttachments.vue:133]: [DEBUG] mail: checking upload size limit 
Object { app: "mail", uid: "greta", existingUploads: 0, toUpload: 20436, limit: "0", newTotal: 20436 }

Comment (GretaD) [reply src/components/ComposerAttachments.vue:133]: also, sorry i forgot to include in the first comment, when i click New Message is shows this: 
```
[Vue warn]: Invalid prop: type check failed for prop "uploadSizeLimit". Expected Number with value 0, got String with value "0".

found in

---> <ComposerAttachments> at src/components/ComposerAttachments.vue
```

Comment (ChristophWurst) [reply src/components/ComposerAttachments.vue:133]: We need to cast. On it.

Comment (ChristophWurst) [reply src/components/ComposerAttachments.vue:133]: Please try again.

**Comment by st3iny** [comment]
The check is not done when adding a file from the cloud as a direct attachment. I looked through the code of the filepicker and it only returns a path. Implementing this would require some form of api call to get the file size. Do we want this?

**Comment by ChristophWurst** [comment]
> The check is not done when adding a file from the cloud as a direct attachment. I looked through the code of the filepicker and it only returns a path. Implementing this would require some form of api call to get the file size. Do we want this?

It's kind of inevitable, I think. Or at least I don't know an alternative off the top of my head :wink: 
If we one day have a modern file picker we can maybe return file info objects instead of just paths. That would help a lot.

**Comment by GretaD** [comment]
When i upload a file that is above limit, on UI it doesnt show any warning, just the console error
```
[ERROR] mail: could not upload all attachments 
app: "mail"
error: Error: Request failed with status code 500
uid: "greta"
```

**Review by st3iny — APPROVED** [review]
Works for me. Cloud attachments are now checked as well.

**Comment by ChristophWurst** [comment]
> error: Error: Request failed with status code 500

@GretaD do you get the same response as @st3iny in https://github.com/nextcloud/mail/issues/4217?
What does Nextcloud log say?

**Comment by ChristophWurst** [comment]
Merging in the hope that the attachments bug is not related to my change :crossed_fingers: 

@GretaD pleast test again and check your logs. Then open a ticket with the details.

**Pull request merged**